lib/Conversion/TritonGPUToLLVM/ConvertLayoutOpToLLVM/SharedToDotOperandMMAv2.cpp

```diff
@@ -138,7 +141,13 @@ MMA16816SmemLoader::computeLdmatrixMatOffs(Value warpId, Value lane,
 
   SmallVector<Value> offs(numPtrs);
   Value phase = urem(udiv(sOffInMat, i32_val(perPhase)), i32_val(maxPhase));
-  Value sOff = add(sOffInMat, mul(sMatOff, i32_val(sMatShape)));
+  // To prevent out-of-bound access of B when wpt * 16 > tile_size.
+  // In such a case, we need to wrap around the offset of B.
+  // |0 1 2 3 0 1 2 3| -> | 0(0) 1(1) 2(2) 3(3) |
+  // |0 1 2 3 0 1 2 3|    | 0(0) 1(1) 2(2) 3(3) |
+  //          ~~~~~~~ out-of-bound access
+  Value sOff = urem(add(sOffInMat, mul(sMatOff, i32_val(sMatShape))),
```

Comment:
The `urem` here doesn't change the underlying ptx instructions at least for the matmul tutorial.